bigframes/operations/semantics.py

```diff
@@ -194,6 +194,88 @@ def agg(
 
         return df[column]
 
+    def cluster_by(
+        self,
+        column: str,
+        output_column: str,
+        model,
+        n_clusters: int = 5,
+    ):
+        """
+        Clusters data based on the semantic similarity of text within a specified column.
+
+        This method leverages a language model to generate text embeddings for each value in
+        the given column. These embeddings capture the semantic meaning of the text.
+        The data is then grouped into `n` clusters using the k-means clustering algorithm,
+        which groups data points based on the similarity of their embeddings.
+
+        **Examples:**
+
+            >>> import bigframes.pandas as bpd
+            >>> bpd.options.display.progress_bar = None
+            >>> bpd.options.experiments.semantic_operators = True
+
+            >>> import bigframes.ml.llm as llm
+            >>> model = llm.TextEmbeddingGenerator()
+
+            >>> df = bpd.DataFrame({
+            ...     "Product": ["Smartphone", "Laptop", "T-shirt", "Jeans"],
+            ... })
+            >>> df.semantics.cluster_by("Product", "Cluster ID", model, n_clusters=2)
+                    Product  Cluster ID
+            0    Smartphone           2
+            1        Laptop           2
+            2       T-shirt           1
+            3         Jeans           1
```

Comment:
is this deterministic - or are cluster centroids initialized randomly?

Reply:
Though initial tests suggest consistent results, I'm not fully convinced. I'll continue monitoring the test outcomes. If inconsistencies arise, I'll adjust the verification process to focus solely on the number of unique IDs.